app/kumactl/cmd/apply/apply.go

```diff
@@ -38,7 +40,16 @@ func NewApplyCmd(pctx *kumactl_cmd.RootContext) *cobra.Command {
 			if ctx.args.file == "" || ctx.args.file == "-" {
 				b, err = ioutil.ReadAll(cmd.InOrStdin())
 			} else {
-				b, err = ioutil.ReadFile(ctx.args.file)
+				if strings.HasPrefix(ctx.args.file, "http") {
+					resp, err := http.Get(ctx.args.file)
```

Comment:
Use HTTP client with timeouts. Otherwise, a command can be stuck forever.
See: kuma/pkg/util/http/client.go